specification/appconfiguration/resource-manager/Microsoft.AppConfiguration/preview/2024-09-01-preview/appconfiguration.json

```diff
@@ -0,0 +1,3260 @@
+            "schema": {
+              "$ref": "#/definitions/RegenerateKeyParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/ApiKey"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ConfigurationStores_RegenerateKey": {
+            "$ref": "./examples/ConfigurationStoresRegenerateKey.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/generateSasToken": {
+      "post": {
+        "tags": [
+          "ConfigurationStores"
+        ],
+        "description": "Generates a SAS token for scoped, read-only access of the specified configuration store.",
+        "operationId": "ConfigurationStores_GenerateSasToken",
+        "parameters": [
+          {
+            "$ref": "#/parameters/SubscriptionIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ResourceGroupNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ConfigStoreNameParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "name": "sasTokenGenerationParameters",
+            "in": "body",
+            "description": "The object containing information for the SAS token generation request.",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/SasTokenGenerationParameters"
+            }
+          }
+        ],
+        "responses": {
+          "200": {
+            "description": "The request was successful; the request was well-formed and received properly.",
+            "schema": {
+              "$ref": "#/definitions/SasTokenGenerationResult"
+            }
+          },
+          "default": {
+            "description": "Error response describing why the operation failed",
+            "schema": {
+              "$ref": "#/definitions/ErrorResponse"
+            }
+          }
+        },
+        "x-ms-examples": {
+          "ConfigurationStores_GenerateSasToken": {
+            "$ref": "./examples/ConfigurationStoresGenerateSasToken.json"
+          }
+        }
+      }
+    },
+    "/subscriptions/{subscriptionId}/resourceGroups/{resourceGroupName}/providers/Microsoft.AppConfiguration/configurationStores/{configStoreName}/resetSasKind": {
+      "post": {
+        "tags": [
+          "ConfigurationStores"
+        ],
+        "description": "Reset all SAS tokens of a given kind (primary/secondary).",
```

Comment:
Reset SAS kind to invalidate all previously generated SAS tokens of the specified kind.

Reply:
Updated